File: app/src/main/java/top/txwgoogol/delayedfragment/BasePageAdapter.java
package top.txwgoogol.delayedfragment;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.List;

public class BasePageAdapter extends FragmentStatePagerAdapter {

    private List<Fragment> fragments;
    private List<String> titles;

    public BasePageAdapter(FragmentManager fm) {
        super(fm);
    }

    public BasePageAdapter(FragmentManager fm, List<Fragment> fragments) {
        super(fm);
        this.fragments = fragments;
    }

    public BasePageAdapter(FragmentManager fm, List<Fragment> fragments, List<String> titles) {
        super(fm);
        this.fragments = fragments;
        this.titles = titles;
    }

    @Override
    public Fragment getItem(int i) {
        return fragments.get(i);
    }

    @Override
    public int getCount() {
        return fragments == null ? 0 : fragments.size();
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return titles.get(position);
    }

}

File: delay-fragment-kotlin/app/src/main/java/com/colin/delay/base/BasePageAdapter.kt
package com.colin.delay.base

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class BasePageAdapter : FragmentStatePagerAdapter {

	private var fragments: List<Fragment>? = null
	private var titles =  mutableListOf<String>()

	constructor(fm: FragmentManager) : super(fm) {}

	constructor(fm: FragmentManager, fragments: List<Fragment>) : super(fm) {
		this.fragments = fragments
	}

	constructor(fm: FragmentManager, fragments: List<Fragment>, titles: MutableList<String>) : super(fm) {
		this.fragments = fragments
		this.titles = titles
	}

	override fun getItem(i: Int): Fragment {
		return fragments!![i]
	}

	override fun getCount(): Int {
		return if (fragments == null) 0 else fragments!!.size
	}

	override fun getPageTitle(position: Int): CharSequence? {
		return titles[position]
	}

}